src/__tests__/types.test.ts

```diff
@@ -214,3 +214,13 @@ describe("withDefault (composed partial)", () => {
     }
   });
 });
+
+describe(`Head<T>`, () => {
+  it("should map correct type for an array", () => {
+    const c = [1, "two", false];
```

Comment:
Add `as const` to make it be treated as a struct and not and array.

```suggestion
    const c = [1, "two", false] as const;
```

Reply:
Brilliant! Thanks